pkg/server/handler.go

```diff
@@ -253,6 +257,86 @@ func WithRequestIdentity(handler http.Handler) http.Handler {
 	})
 }
 
+type privilege int
+
+const (
+	superPrivileged privilege = iota
+	priviledged
+	unprivileged
+)
```

Comment:
Maybe it's just me, but I find this super confusing because it creates some weird situations in the code later on. Should we maybe start from `unprivileged`?

I mean e.g. this code snippet:

```go
// Check if user's max privilege is less than or equal to the requested group's privilege
if userMaxPrivilege > requestedPrivilege {
	return false // Deny impersonation if user's privilege is not sufficient
}
```

The comment is basically saying the inverse of the actual if condition. I feel like this will be prone to errors if we have to modify it at some point.

Reply:
cc @sttts since you proposed this change, I think.

Reply:
Commented the same below: invert the order. Makes it simpler.